segment.go

```diff
@@ -247,18 +218,82 @@ func (seg *segment) Write(data []byte) (*ChunkPosition, error) {
 		default: // Middle chunk
 			chunkType = ChunkTypeMiddle
 		}
-		seg.appendChunkBuffer(chunkBuffer, data[dataSize-leftSize:end], chunkType)
+		seg.appendChunkBuffer(buf, data[dataSize-leftSize:end], chunkType)
 
 		leftSize -= chunkSize
 		blockCount += 1
 		currBlockSize = (currBlockSize + chunkSize + chunkHeaderSize) % blockSize
 	}
 
+	position.ChunkSize = blockCount*chunkHeaderSize + dataSize
+
+	return position, padding, nil
+}
+
+// WriteAll write batch data to the segment file.
+func (seg *segment) WriteAll(data [][]byte) ([]*ChunkPosition, error) {
```

Comment:
can u maybe privatize all segment functions that are invoked from within wal's methods? such as WriteAll, Write, Read.. as they arent supposed to be used independently. #22